packages/core/src/apiml/Services.ts

```diff
@@ -0,0 +1,244 @@
+                            ...(profInfo.basePathsByPlugin[config.pluginName] || []),
+                            apiInfo.basePath
+                        ]);
+                    }
+                }
+            }
+        }
+
+        // Find conflicts in profile info array
+        for (const profInfo of profInfos) {
+            // If multiple CLI plug-ins require different base paths for the
+            // same service ID and CLI profile type, we have a conflict because
+            // the plugins expect different base paths and may be incompatible.
+
+            // Find the intersection of base path sets for all CLI plug-ins
+            const basePathSets = Object.values(profInfo.basePathsByPlugin).map(set => new Array(...set));
+            const commonBasePaths = basePathSets.reduce((a, b) => a.filter(x => b.includes(x)));
+            if (commonBasePaths.length === 0) {
+                profInfo.conflictTypes.push("basePaths");
+            } else {
+                profInfo.basePaths = profInfo.basePaths.filter(basePath => commonBasePaths.includes(basePath));
+            }
+            delete profInfo.basePathsByPlugin;
+
+            // If multiple profile infos have the same type, we have a conflict
+            // because we don't know which profile should be the default.
+            if (profInfos.filter(({ profType }) => profType === profInfo.profType).length > 1) {
+                profInfo.conflictTypes.push("profType");
+            }
+        }
+
+        return profInfos;
+    }
+
+    /**
+     * Converts apiml profile information to team config profile objects to be stored in a zowe.config.json file
+     * @param profileInfoList List of apiml profiles
+     * @returns List of config profile objects
+     * @memberof Services
+     */
+    public static convertApimlProfileInfoToProfileConfig(profileInfoList: IApimlProfileInfo[]): IConfig {
+        const configProfile: IConfigProfile = {
+            properties: {},
+            profiles: {}
+        };
+
+        let configDefaults: {[key: string]: string} = {};
+        const conflictingDefaults: {[key: string]: string[]} = {};
+        const configPlugins: Set<string> = new Set<string>();
+
+        const _genCommentsHelper = (key: string, elements: string[]): string => {
+            if (elements == null) return "";
+            return `//"${key}": "${elements.length === 1 ? elements[0] : elements.join('"\n//"' + key + '": "')}"`;
+        }
+
+        profileInfoList.forEach((profileInfo: IApimlProfileInfo) => {
+
+            profileInfo.pluginConfigs.forEach((pluginInfo: IApimlSvcAttrsLoaded) => {
+                configPlugins.add(pluginInfo.pluginName);
+            });
+
+            if (!configDefaults[profileInfo.profType]) {
+                configDefaults[profileInfo.profType] = profileInfo.profName;
+            }
+            if (profileInfo.conflictTypes.includes("profType") && profileInfo.profName !== configDefaults[profileInfo.profType]) {
```

Comment:
This may be a wrong assumption but this `if (profileInfo.conflictTypes.includes("profType") && profileInfo.profName !== configDefaults[profileInfo.profType]))` could be replaced with an `else` and combined with the previous clause.

I do believe this is more explicit.

My only concern is about something like this:
```Javascript
const testCase: IApimlProfileInfo[] = [
  {
    profName: "test2.1",
    profType: "type2",
    basePaths: [
        "test2.1/v1"
    ],
    pluginConfigs: new Set(),
    conflictTypes: []
  },
  {
    profName: "test2.2",
    profType: "type2",
    basePaths: [
        "test2.2/v1"
    ],
    pluginConfigs: new Set(),
    conflictTypes: []
  }
];
```

Clearly two profiles exists of the same type but no profType conflict was detected.

Should an error be thrown in this scenario?
And a test added too 😋 ?

Reply:
Removed `profType` check for a simplified conflict check.
Conflicts are now detected even if there is no conflictTypes provided.